tests/unittests/sources/test_oracle.py

```diff
@@ -917,15 +1060,88 @@ def test_secondary_nic_failure_isnt_blocking(
             side_effect=Exception(),
         ):
             network_config = oracle_ds.network_config
-        assert network_config == m_read_initramfs_config.return_value
-        assert "Failed to parse secondary network configuration" in caplog.text
+        assert network_config == oracle_ds._get_iscsi_config.return_value
+        assert 2 == caplog.text.count(
+            "Failed to parse IMDS network configuration"
+        )
 
-    def test_ds_network_cfg_preferred_over_initramfs(self, _m):
+    def test_ds_network_cfg_preferred_over_initramfs(
+        self, m_get_interfaces_by_mac
+    ):
         """Ensure that DS net config is preferred over initramfs config"""
         config_sources = oracle.DataSourceOracle.network_config_sources
         ds_idx = config_sources.index(NetworkConfigSource.DS)
         initramfs_idx = config_sources.index(NetworkConfigSource.INITRAMFS)
         assert ds_idx < initramfs_idx
 
+    @pytest.mark.parametrize("set_primary", [True, False])
+    def test__add_network_config_from_opc_imds_no_vnics_data(
```

Comment:
Nit: Is the double underscore intentional?

Reply:
Yes, it is. I am following the convention of the file of naming the test with `test_` + `method name`, and this case the method is private, thus the double underscores.